### ISSUE TITLE
Could not find com.adapty.internal:crossplatform:2.3 Android

### ISSUE THREAD

**Issue by VashaMasha** [adaptyteam/AdaptySDK-React-Native#51]
Hi Adapty,

I updated "react-native" from 0.63.5 to 0.71.3 and "react-native-adapty" from 1.3.8 to 2.3.5 (also tried 2.3.2). 
And now android build fails with "Could not find com.adapty.internal:crossplatform:2.3" error. 

```
Could not determine the dependencies of task ':app:processDebugResources'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.adapty.internal:crossplatform:2.3.
     Searched in the following locations:
       - https://oss.sonatype.org/content/repositories/snapshots/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://repo.maven.apache.org/maven2/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - file:/Users/maria/neurocog-app/node_modules/jsc-android/dist/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://dl.google.com/dl/android/maven2/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://www.jitpack.io/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
     Required by:
         project :app > project :react-native-adapty

```

**Comment by divanc** [comment]
Hello, have you tried fetching new dependencies? Via gradle CLI or Android Studio?

**Comment by VashaMasha** [comment]
> Hello, have you tried fetching new dependencies? Via gradle CLI or Android Studio?

My dependencies looks like this

android/build.graddle:
```
dependencies {
        classpath("com.android.tools.build:gradle:7.3.1")
        classpath("com.facebook.react:react-native-gradle-plugin")
        classpath 'com.google.gms:google-services:4.3.15'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.21"
 }
```
 
 android/app/build.graddle:
 ```
dependencies {
    implementation 'com.facebook.android:facebook-android-sdk:latest.release'
    implementation "com.facebook.react:react-native:+"
    implementation platform('com.google.firebase:firebase-bom:28.2.1')
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    implementation project(':@react-native-camera-roll_camera-roll')

    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}")
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }

    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}")

    if (hermesEnabled.toBoolean()) {
        implementation("com.facebook.react:hermes-android")
    } else {
        implementation jscFlavor
    }
}
```

**Comment by divanc** [comment]
It's all looking fine to me. Can you share a little bit more info? 
- Your OS
- React Native version

Can you try in your android folder?
```sh
./gradlew clean
./gradlew build
```

**Comment by VashaMasha** [comment]
./gradlew clean - success
./gradlew build - fail with log (👀):

```
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':lottie-react-native:androidJavadocs'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): '/Users/maria/neurocog-app/node_modules/lottie-react-native/src/android/build/tmp/androidJavadocs/javadoc.options'

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
===========================================
```

**Comment by divanc** [comment]
And your Kotlin version?

**Comment by divanc** [comment]
Stupid of me, you've dropped it above: 1.6.21

**Comment by VashaMasha** [comment]
I use gradle 7.5.1

android/build.gradle
```
buildscript {
    ext {
        buildToolsVersion = "33.0.0"
        minSdkVersion = 24
        compileSdkVersion = 33
        targetSdkVersion = 33
        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
        ndkVersion = "23.1.7779620"
    }
    repositories {
        google()
        mavenCentral()
    }
    dependencies {
        classpath("com.android.tools.build:gradle:7.3.1")
        classpath("com.facebook.react:react-native-gradle-plugin")
        classpath 'com.google.gms:google-services:4.3.15'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.21"
    }
}
```

android/app/build.gradle:

```
apply plugin: "com.android.application"
apply plugin: "com.facebook.react"
apply plugin: 'com.google.gms.google-services'


import com.android.build.OutputFile

/**
* This is the configuration block to customize your React Native Android app.
* By default you don't need to apply any configuration, just uncomment the lines you need.
*/


apply from: "../../node_modules/@sentry/react-native/sentry.gradle"

/**
*  Set this to true to create four separate APKs instead of one,
* one for each native architecture. This is useful if you don't
* use App Bundles (https://developer.android.com/guide/app-bundle/)
* and want to have separate APKs to upload to the Play Store.
*/
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
 */
def enableProguardInReleaseBuilds = false

/**
 * The preferred build flavor of JavaScriptCore (JSC)
 *
 * For example, to use the international variant, you can use:
 * `def jscFlavor = 'org.webkit:android-jsc-intl:+'`
 *
 * The international variant includes ICU i18n library and necessary data
 * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
 * give correct results when using with locales other than en-US. Note that
 * this variant is about 6MiB larger per architecture than default.
 */
def jscFlavor = 'org.webkit:android-jsc:+'

/**
* Private function to get the list of Native Architectures you want to build.
* This reads the value from reactNativeArchitectures in your gradle.properties
* file and works together with the --active-arch-only flag of react-native run-android.
*/
def reactNativeArchitectures() {
    def value = project.getProperties().get("reactNativeArchitectures")
    return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
}

android {
    ndkVersion rootProject.ext.ndkVersion

    compileSdkVersion rootProject.ext.compileSdkVersion

    namespace "com.testApp"
    defaultConfig {
        applicationId "com.testApp"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 267
        versionName "2.2.11"
        missingDimensionStrategy ('store', 'play')
        multiDexEnabled true
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include (*reactNativeArchitectures())
        }
    }
    signingConfigs {
        release {
            if (project.hasProperty('MYAPP_RELEASE_STORE_FILE')) {
                storeFile file(MYAPP_RELEASE_STORE_FILE)
                storePassword MYAPP_RELEASE_STORE_PASSWORD
                keyAlias MYAPP_RELEASE_KEY_ALIAS
                keyPassword MYAPP_RELEASE_KEY_PASSWORD
            }
        }
    }

    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // https://developer.android.com/studio/build/configure-apk-splits.html
            // Example: versionCode 1 will generate 1001 for armeabi-v7a, 1002 for x86, etc.
            def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        defaultConfig.versionCode * 1000 + versionCodes.get(abi)
            }

        }
    }

    dexOptions {
       javaMaxHeapSize "3g"
    }
}

dependencies {
    implementation 'com.facebook.android:facebook-android-sdk:latest.release'
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation platform('com.google.firebase:firebase-bom:28.2.1')
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    implementation project(':@react-native-camera-roll_camera-roll')

    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}")
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }

    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}")

    if (hermesEnabled.toBoolean()) {
        implementation("com.facebook.react:hermes-android")
    } else {
        implementation jscFlavor
    }
    constraints {
        implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.8.0") {
            because("kotlin-stdlib-jdk7 is now a part of kotlin-stdlib")
        }
        implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.8.0") {
            because("kotlin-stdlib-jdk8 is now a part of kotlin-stdlib")
        }
    }
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
```

**Comment by VashaMasha** [comment]
Summary. I tried:

1. Without react-native-adapty:
Project builds (both debug and release) and launces, all good

2. With react-native-adapty 1.3.8 (i used this version in previous react-native version 0.63.5 and kotlin version 1.5.31:
Project builds (both debug and release). If adapty methods not used, only lib installed, app launches and works fine. 
If i try to use adapty methods - app crashes with error (log from sentry, I can't see it in console)
```
NoSuchMethodError
No virtual method queryPurchases(Ljava/lang/String;)Lcom/android/billingclient/api/Purchase$PurchasesResult; in class Lcom/android/billingclient/api/BillingClient; or its super classes (declaration of 'com.android.billingclient.api.BillingClient' appears in /data/app/~~J8w4aOAJrA8MQRy-t_OxPA==/com.testApp-zV_QROvD5u_RS6CloKXVWA==/base.apk)
```

3. With react-native-adapty 2.3.2+:
Builds fail (both debug and release) with error:
```
* What went wrong:
Could not determine the dependencies of task ':app:processDebugResources'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.adapty.internal:crossplatform:2.3.
     Searched in the following locations:
       - https://oss.sonatype.org/content/repositories/snapshots/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://repo.maven.apache.org/maven2/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - file:/Users/maria/testApp-app/node_modules/jsc-android/dist/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://dl.google.com/dl/android/maven2/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://www.jitpack.io/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
     Required by:
         project :app > project :react-native-adapty

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 17s

    at makeError (/Users/maria/testApp-app/node_modules/execa/index.js:174:9)
    at /Users/maria/testApp-app/node_modules/execa/index.js:278:16
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async runOnAllDevices (/Users/maria/testApp-app/node_modules/react-native/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/runOnAllDevices.js:82:7)
    at async Command.handleAction (/Users/maria/testApp-app/node_modules/@react-native-community/cli/build/index.js:108:9)
info Run CLI with --verbose flag for more details.
```

**Comment by divanc** [comment]
I currently have no idea! Working on it...

**Comment by divanc** [comment]
Hey, @VashaMasha, can you try `v2.7.0-beta` version? I've changed the way Gradle should look for a local maven repo

**Comment by divanc** [comment]
@VashaMasha nevermind `v2.3.7-beta`, how about `v2.3.10`?

**Comment by VashaMasha** [comment]
> @VashaMasha nevermind `v2.3.7-beta`, how about `v2.3.10`?

My colleague created a new project and transferred all the functionality of the application to it - everything is fine regardless of the library (tried 2.3.6 and 2.3.7-beta). I tried v2.3.7-beta and v2.3.10 in my branch (without creating a new project) and got the same error.
It can be concluded that the problem was in the project, and not in the library, but I don't understand where exactly. New project has different structure in the "android" folder (in the screenshot, the old one is on the left, the new one is on the right). But if this is the problem then why does the error occur only when trying to use new adapty. And without adapty or with its older version the same project builds fine. 

<img width="470" alt="Screenshot 2023-02-27 at 14 04 49" src="https://user-images.githubusercontent.com/36168154/221839853-583b915c-9341-42a6-a5ba-cac2ec0807d7.png">

So in conclusion - everything works with the new project
Divanc, thank you so mush for your time and your help!

**Comment by divanc** [comment]
I'm here to help with your problems regardless! Gonna check your info a little later

**Comment by divanc** [comment]
@VashaMasha ok, it took me longer, than I expected. I'm sorry!

If you are still facing the issue, what you can do is manually point to this dependency. It is one of mine, so I know where to find it:

In your `android/build.gradle` find `repositories` and add the following:

```groovy

repositories {
    mavenCentral()
    google()
    maven { // This line should fix the problem
        url "$rootDir/../node_modules/react-native-adapty/lib/android/localMaven"
    }
}
```

**Comment by divanc** [comment]
@VashaMasha, I'm going to close this issue as I believe the answer above should fix the problem. If it persists and you look for further help please re-open it

**Comment by dualism98** [comment]
@divanc, hey!) I tried your solution. And seems it should work, but I'm getting this error:
```
* What went wrong:
Could not determine the dependencies of task ':app:processDebugResources'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.adapty.internal:crossplatform:2.3.
     Searched in the following locations:
       - https://oss.sonatype.org/content/repositories/snapshots/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://repo.maven.apache.org/maven2/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - file:/Users/khramushin/Desktop/ai-chat-ios-romantic/node_modules/jsc-android/dist/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://dl.google.com/dl/android/maven2/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
       - https://www.jitpack.io/com/adapty/internal/crossplatform/2.3/crossplatform-2.3.pom
     Required by:
         project :app > project :react-native-adapty

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s

```

**Comment by divanc** [comment]
@dualism98, sorry, skipped your comment due to closed issue. If you are still experiencing this issue, can you confirm that such path exists?
`$rootDir/../node_modules/react-native-adapty/lib/android/localMaven`

so it is one dir above from your android folder and then into node_modules
